synchronization/src/cyface/java/de/cyface/synchronization/CyfaceAuthenticator.java

```diff
@@ -245,58 +234,18 @@ public Bundle hasFeatures(AccountAuthenticatorResponse response, Account account
      * @throws MalformedURLException If no protocol is specified, or an unknown protocol is found, or spec is null.
      * @throws ServerUnavailableException When there seems to be no server at the given URL.
      * @throws NetworkUnavailableException When the network used for transmission becomes unavailable.
-     * @throws TooManyRequestsException When the server returns {@link HttpConnection#HTTP_TOO_MANY_REQUESTS}
+     * @throws TooManyRequestsException When the server returns {@code Uploader#HTTP_TOO_MANY_REQUESTS}
      * @throws ForbiddenException E.g. when there is no actual API running at the URL
      * @throws UnexpectedResponseCode When the server returns an unexpected response code
      * @throws AccountNotActivated When the user account is not activated
      */
     private String login(final @NonNull String username, final @NonNull String password)
             throws ServerUnavailableException, MalformedURLException, SynchronisationException, UnauthorizedException,
-            NetworkUnavailableException, TooManyRequestsException, HostUnresolvable, ForbiddenException, UnexpectedResponseCode, AccountNotActivated {
-        Log.v(TAG, "Logging in to get new authToken");
-
-        // Load authUrl
-        final SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
-        final String url = preferences.getString(SyncService.SYNC_ENDPOINT_URL_SETTINGS_KEY, null);
-        if (url == null) {
-            throw new IllegalStateException(
-                    "Server url not available. Please set the applications server url preference.");
-        }
-        final URL authUrl = new URL(http.returnUrlWithTrailingSlash(url) + "login");
-
-        // Generate login payload
-        final JSONObject loginPayload = new JSONObject();
-        try {
-            loginPayload.put("username", username);
-            loginPayload.put("password", password);
-        } catch (final JSONException e) {
-            throw new IllegalArgumentException(e);
-        }
+            NetworkUnavailableException, TooManyRequestsException, HostUnresolvable, ForbiddenException,
+            UnexpectedResponseCode, AccountNotActivated {
```

Comment:
Would it be possible to wrap all these exceptions in a common one (like LoginFailed), reducing the footprint of this function, but still keeping the information? The Actual stack trace should still persist, wouldn't it?

Reply:
I've simplified the API of `Authenticator` and `Uploader` in `uploader` library to return `Login-/Registration-/UploadFailed` as exception, but in the SDK in `CyfaceAuthenticator` we still need some code to pass the exception type to the `ErrorHandler`:

```
} catch (final LoginFailed e) {
            var cause = e.getCause();
            if (cause instanceof ServerUnavailableException || cause instanceof ForbiddenException) {
                sendErrorIntent(context, SERVER_UNAVAILABLE.getCode(), e.getMessage());
                throw new NetworkErrorException(e);
            } else if (cause instanceof MalformedURLException) {
                sendErrorIntent(context, MALFORMED_URL.getCode(), e.getMessage());
                throw new NetworkErrorException(e);
            } else if (cause instanceof SynchronisationException) {
                sendErrorIntent(context, SYNCHRONIZATION_ERROR.getCode(), e.getMessage());
```

Maybe there is a better solution for error handling, but I don't want to refactor the SDK as the purpose of the PR is just to extract the auth/upload functions to the `uploader` library with a simple API for 3rd parties.

I think we're in a good spot now, with the current `uploader` interface:
```
public abstract fun upload(
  jwtToken: kotlin.String,
  metaData: de.cyface.model.RequestMetaData,
  file: java.io.File,
  progressListener: de.cyface.uploader.UploadProgressListener
): de.cyface.uploader.Result
// now only throws `UploadFailed` wrapper exception

```